### ISSUE TITLE
driver.findElements(By.id("id")) doesn't work. iOS 7.x/iOS 8.x

### ISSUE THREAD

**Issue by TikhomirovSergey** [appium/appium#4180]
This issue was found by @mariusbob. 
The starting of this tread:  https://github.com/appium/appium/issues/4123
It was hard to define the reason of this error.

So...
Appium 1.3.3
Mac OS X 9.5
XCode 6.01
iOS 8.0
App: [Albertsons.zip](http://mariusbob.ro/Albertsons.zip)

This code works cool :+1: 

``` java
driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
driver.findElement(By.id("register.firstName.placeholder")).sendKeys("bla-bla");
```

But 

``` java
driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
driver.findElements(By.id("register.firstName.placeholder")).get(0).sendKeys("bla-bla");
```

fails :( 
It is known after conversation with @mariusbob that ""register.firstName.placeholder" is an App string identifier

Stacktrace:

> java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
>     at java.util.ArrayList.rangeCheck(ArrayList.java:653)
>     at java.util.ArrayList.get(ArrayList.java:429)
>     at 
> 
>   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke(Method.java:483)
>     at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
>     at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
>     at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
>     at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
>     at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
>     at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
>     at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
>     at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
>     at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
>     at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
>     at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
>     at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
>     at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
>     at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
>     at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
>     at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)
>     at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
>     at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
>     at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
>     at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
>     at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)

So there is empty list but was expected list which contains 1 element

Logs:
https://gist.github.com/TikhomirovSergey/3e511103cafbea82e7e2

**Comment by archanaravilla** [comment]
Has this issue been fixed for ios annotations? @TikhomirovSergey

**Comment by TikhomirovSergey** [comment]
This is not the annotations issue. I can check it later. But I expect that problem is still here. Milestone is 1.3.6.

**Comment by mariusbob** [comment]
Hi,
I'm seeing this issue still opened. Was it fixed in 1.3.7? 
Thanks.

**Comment by jlipps** [comment]
This should be fixed in the 1.5 rewrite
